designs/2018-processors-improvements/README.md

```diff
@@ -0,0 +1,202 @@
+- Start Date: 2018-11-20
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Nicholas C. Zakas (@nzakas)
+
+# Processors Improvements
+
+## Summary
+
+This proposal provides a way to explicitly define which processor(s) to use for different files inside of configuration. It also allows the chaining of multiple processors to fully process a file.
+
+## Motivation
+
+The current processor design is inverted from the way ESLint configuration works in all other cases. Plugins must specify which file extension(s) a processor should be applied to, and that setting is honored as soon as the plugin is loaded into ESLint. End-users do not have the option of disabling the processor or changing which files it should apply to.
+
+Additionally, there are complex conditions such as Vue code inside of a Markdown file (see [#11035](https://github.com/eslint/eslint/issues/11035)) that are currently very difficult because it requires coordination between plugins directly rather than changes in configuration.
+
+The goals of this proposal are:
+
+1. Allow end-users to configure which processors should be used for which files.
+2. Allow chaining of multiple processors for complex conditions, such as Vue code inside of a Markdown file.
+
+## Detailed Design
+
+Design Summary:
+
+1. Require plugins to export named processors instead of/in addition to using file extensions.
+1. Add a `processors` option to configuration.
+1. Use `overrides` in configuration to target the files to use processors.
+1. Apply multiple processors to target files using configuration.
+
+Definitions:
+
+* **Extension-named processor** is the original processor design, where a processor is named only by the file extension to which the process should apply.
+* **Named processor** is introduced in this design, where a processor is given a name just like you would name a rule. The processor is not tied to a specific file extension by default.
```

Comment:
Do we want to state that a named processor must not begin with a dot (to avoid ambiguity with extension-named processor)? And/or do we want to impose a stronger restriction?

Reply:
Yes, that's a good restriction to add in.